src/tests/ftest/util/test_utils_pool.py

```diff
@@ -532,6 +532,49 @@ def check_pool_daos_space(self, s_total=None, s_free=None):
             for index, item in enumerate(val)]
         return self._check_info(checks)
 
+    def check_free_space(self, expected_scm=None, expected_nvme=None,
+                         timeout=30):
+        """Check pool free space with expected value.
+        Args:
+            expected_scm (int, optional): pool expected SCM free space.
+            expected_nvme (int, optional): pool expected NVME free space.
+            timeout(int, optional): time to fail test if it could not match
+                expected values.
+        Note:
+            Arguments may also be provided as a string with a number preceded
+            by '<', '<=', '>', or '>=' for other comparisons besides the
+            default '=='.
+        """
+        if not expected_scm and not expected_nvme:
+            self.log.error("at least one space parameter must be specified")
+            return False
+
+        done = False
+        scm_fs = 0
+        nvme_fs = 0
+        start = time()
+        scm_index, nvme_index = 0, 1
+        while time() - start < timeout:
```

Comment:
Optional: we can check `done` here instead of at L568-L569:
```suggestion
        while (time() - start < timeout) and not done:
```